tests/rptest/tests/retention_policy_test.py

```diff
@@ -443,3 +461,62 @@ def cloud_log_size() -> int:
                    timeout_sec=10,
                    backoff_sec=2,
                    err_msg=f"Too many bytes in the cloud")
+
+    @cluster(num_nodes=3)
+    def test_cloud_time_based_retention(self):
+        """
+        Test that retention is enforced in the cloud log by checking
+        the total size of segments in the manifest.
+        """
+        retention_bytes = 10 * self.segment_size
+        total_bytes = 20 * self.segment_size
+
+        topic = TopicSpec(name=self.topic_name,
+                          partition_count=1,
+                          replication_factor=3,
+                          cleanup_policy=TopicSpec.CLEANUP_DELETE)
+
+        self.redpanda.set_cluster_config(
+            {
+                "cloud_storage_enable_remote_write": True,
+                "cloud_storage_housekeeping_interval_ms": 100
+            },
+            expect_restart=True)
+
+        self.rpk.create_topic(topic=topic.name,
+                              partitions=topic.partition_count,
+                              replicas=topic.replication_factor,
+                              config={
+                                  "cleanup.policy": topic.cleanup_policy,
+                              })
+
+        produce_until_segments(self.redpanda,
```

Comment:
Given that our ultimate success condition will be for the segment count to go to zero, this test could probably be made a bit more flexible by not using exact segment counts at all?  e.g. just produce a set quantity of data in bytes, then wait for the number of segments in cloud to match the number on local disk, then proceed to wait for the number in cloud to hit zero after setting a short retention time.

Reply:
Good proposal. I changed to doing that and updated the test description to reflect the new approach.